tequilapi/endpoints/auth.go

```diff
@@ -113,7 +117,10 @@ func (api *authenticationAPI) Authenticate(httpRes http.ResponseWriter, httpReq
 //     description: Internal server error
 //     schema:
 //       "$ref": "#/definitions/ErrorMessageDTO"
-func (api *authenticationAPI) Login(httpRes http.ResponseWriter, httpReq *http.Request, _ httprouter.Params) {
+func (api *authenticationAPI) Login(c *gin.Context) {
+	httpReq := c.Request
+	httpRes := c.Writer
```

Comment:
Well this seems a bit backwards to me. From a quick search it seems that most of the `SendError` or `WriteAsJSON` are only used in tequila and they all write JSON. So in theory functions in `tequilapi/utils/utils.go` could just accept interface:
```
type JSONErrorWriter interface {
	JSON(code int, body interface{})
}
```

And you could just pass `c` into those funcs. I guess this could also be improvement num 2 🤷🏻

Reply:
Like it says in the commit description: Quick & Dirty, I would like to throw away those utils and maybe swap them with a dependency to standardize this across all projects that use gin. So I would wish to leave this for another iteration :) this PR is large as it is.

Reply:
Sure no problem, just pointing an issue to open after this is merged for the future.